### PR TITLE
Allow wildcard pattern debugging

### LINKED ISSUE #17026: use re 'debug' ignored for uniprop embedded regexps
Migrated from [rt.perl.org#134150](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=134150) (status was 'new')

Searchable as RT134150$

### PR DESCRIPTION
**Pull request by khwilliamson**
This fixes #17026

Patterns can now have subpatterns, called wildcards by Unicode.  Prior
to this commit debugging info was only available under DEBUGGING builds
with -Drv.  This commit adds a new keyword, WILDCARD for
'use re qw(Debug ...)' so that whatever other debugging options have
been turned on will show up when a wildcard subpattern is compiled.

The output of this may be voluminous, which is why you have to ask for
it specifically.  Or, the EXTRA option turns it on along with several
other things.

### CODE REVIEW

**Comment by khwilliamson** [comment]
I'm uncertain about this commit, so putting it up for review.

Compiling a Unicode property wildcard, or subpattern, is accomplished by recursively calling the compilation function when such a subpattern is encountered, and then matching it against the known Unicode-specified legal values, all while normal compilation of the rest of the main pattern is suspended.  That is unchanged from when this was introduced in 5.30.

To allow debugging from 'use re qw(Debug...)', this commit introduces a new flag to enable it for these subpatterns (normally someone won't be much interested in this detail of compilation,  as it is all behind the scenes; and the output can be voluminous), and then if the flag isn't set, it wraps the calls to compile and execute with turning off the debugging flags, and then restoring them afterwards.

My concern is that these calls can croak,a dn the flags wouldn't be restored.  That should only matter in the case of an eval or perhaps embedded perls.  Is my concern well-founded?   And if so, are there ways of causing these flags to be restored?   There are two cases.  Under control of -Dr, the flag is PL_debug; under control of 'use re' it is ${^RE_DEBUG_FLAGS}.  That should get restored anyway, I should think, upon departure of lexical scope.

File: regcomp.c [hunk 1]
```diff
@@ -23013,23 +23013,54 @@ S_get_extended_utf8_msg(pTHX_ const UV cp)
...
+        }
+    }
+#  endif
+
+    subpattern_re = re_op_compile_wrapper(sv_2mortal(newSVpvn(subpattern, len)),
                                         /* Like in op.c, we copy the compile
                                          * time pm flags to the rx ones */
                                         (flags & RXf_PMf_COMPILETIME), flags);
+#  ifdef DEBUGGING
+    if (turned_off_PL_debug) {
```
Comment (hvds) [inline regcomp.c:23057]: Is this "undo the changes we made" pattern safe when the call above dies? Do we have a test for that?
